src/rafter_consensus_fsm.erl

```diff
@@ -841,25 +823,39 @@ become_candidate(#state{term=CurrentTerm, me=Me}=State0) ->
     _ = request_votes(State3),
     State3.
 
-become_leader(#state{me=Me, term=Term, init_config=InitConfig}=State) ->
-    NewState = State#state{leader=Me,
+become_leader(#state{me=Me, term=Term, config=Config, init_config=InitConfig, timer=Timer}=State) ->
+    NewState0 = State#state{leader=Me,
                            responses=dict:new(),
                            followers=initialize_followers(State),
                            send_clock = 0,
                            send_clock_responses = dict:new(),
-                           read_reqs = orddict:new()},
+                           read_reqs = orddict:new(),
+                           timer = Timer},
 
     case InitConfig of
         complete ->
             %% Commit a noop entry to the log so we can move the commit index
             Entry = #rafter_entry{type=noop, term=Term, cmd=noop},
-            append(Entry, NewState);
-        _ ->
-            %% First entry must always be a config entry
-            NewState
+            append(Entry, NewState0);
+        undefined ->
```

Comment:
I was a little confused by this, but I have got a sequence of steps which will consistently enter this clause:

1. start three nodes with zero history on them, e.g.
    1. rm -f data/*
    2. bin/start-node one
    3. bin/start-node two
    4. bin/start-node three
2. At one of the nodes, issue a rafter:set_config(ThisNode, Nodes)
3. When the config is stable, kill the leader node

In this situation, after the config becomes stable, `init_config` on the leader is `complete` on the leader, but `undefined` on both of the others.

Sticking an `io:format` or similar in the undefined clause shows that it gets entered.

Hope this helps explain why this clause is here! Although, perhaps it should be solved another way.